twifbeddit-app/backend/src/posts.js

```diff
@@ -24,24 +24,50 @@ exports.POST = async (_, event) => {
     }
 };
 
-// TODO: pagination
+// TODO: Test this. 
 exports.GET = async (_, event) => {
-    if (event.queryStringParameters != null)
-        console.error("Parameters currently ignored");
-    const username = await get_user_from_header(event.headers);
-    const user = await User.findOne({username: username});
-    var posts = await Post.find().where('topic').in(user.following).exec();
-    posts.forEach((post, index, arr) => {
-        if (post.anonymous) post.author = null;
-        post.anonymous = undefined;
-        post.__v = undefined;
-        arr[index] = post;
-    });
+    if (event.queryStringParameters == null) {
+        // get posts from the specific user
 
-    return {
-        'statusCode': 200,
-        'body': JSON.stringify({
-            posts: posts
-        })
+        const username = await get_user_from_header(event.headers);
+        const user = await User.findOne({username: username});
+        var posts = await Post.find().where('topic').in(user.followed_topics).limit(8).sort({ createdAt: -1}).exec();
+        posts.forEach((post, index, arr) => {
+            if (post.anonymous) post.author = null;
+            post.anonymous = undefined;
+            post.__v = undefined;
+            arr[index] = post;
+        });
+    
+        return {
+            'statusCode': 200,
+            'body': JSON.stringify({
+                posts: posts
+            })
+        }
+    } else if (event.queryStringParameters.topic != null) {
```

Comment:
This doesn't work. It won't have the value `null`. It just wouldn't be in `queryStringParameters`. You want to compare to `undefined`.

Also rework this. Don't make it an else if. Just after `if (event.queryStringParameters == null)` Do `if (event.queryStringParameters.topic == undefined) return {'statusCode': 404}` Then what you have here. That's cleaner

Reply:
Fixed